test/e2e_node/lifecycle_hook_test.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/intstr"
+	"k8s.io/kubernetes/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = framework.KubeDescribe("Container Lifecycle Hook", func() {
+	f := framework.NewDefaultFramework("container-lifecycle-hook")
+	var podClient *framework.PodClient
+	var file string
+	const podWaitTimetout = 2 * time.Minute
+
+	testPodWithHook := func(podWithHook *api.Pod) {
+		podCheckStart := getTestPod("pod-check-start",
+			// Wait until the file is created.
+			[]string{"sh", "-c", fmt.Sprintf("while [ ! -e %s ]; do sleep 1; done", file)},
+		)
+		podCheckStop := getTestPod("pod-check-stop",
+			// Wait until the file is deleted.
+			[]string{"sh", "-c", fmt.Sprintf("while [ -e %s ]; do sleep 1; done", file)},
+		)
+		By("create the pod with lifecycle hook")
+		podClient.CreateSync(podWithHook)
+		By("create the poststart check pod")
+		podClient.Create(podCheckStart)
+		By("wait for the poststart check pod to success")
+		podClient.WaitForSuccess(podCheckStart.Name, podWaitTimetout)
+		By("delete the pod with lifecycle hook")
+		podClient.DeleteSync(podWithHook.Name, api.NewDeleteOptions(15), podWaitTimetout)
+		By("create the prestop check pod")
+		podClient.Create(podCheckStop)
+		By("wait for the prestop check pod to success")
+		podClient.WaitForSuccess(podCheckStop.Name, podWaitTimetout)
+	}
+
+	BeforeEach(func() {
+		podClient = f.PodClient()
+		file = "/tmp/test-" + string(uuid.NewUUID())
+	})
+
+	Context("when create a pod with poststart/prestop exec hook", func() {
```

Comment:
Just a suggestion. It may be simpler/easier to debug if we create different tests for post-start and pre-stop hooks. In your test, if post-start hook fails, pre-stop hook is also doomed.